libcaf_core/src/detail/base64.cpp

```diff
@@ -41,22 +41,23 @@ void encode_impl(string_view in, Storage& out) {
     for (auto x : buf)
       out.push_back(static_cast<value_type>(encoding_tbl[x]));
   };
+  auto end = [](std::string_view str) { return str.data() + str.size(); };
   // Iterate the input in chunks of three bytes.
-  auto i = in.begin();
-  for (; std::distance(i, in.end()) >= 3; i += 3)
+  auto i = in.data();
+  for (; std::distance(i, end(in)) >= 3; i += 3)
```

Comment:
Is there a technical reason to use `std::end` here over the member function? Otherwise please stick to the member function for consistency.

```suggestion
  for (; std::distance(i, in.end()) >= 3; i += 3)
```

Reply:
Yes. I don't use `std::end` here but the functor `end` from line 44. Once again a windows problem where I couldn't interpret the `std::string_view::iterator` from MSVC as `char*`.. Hence I had to improvise and use the actual `data()` ptr and calculate the size for the routine.

Reply:
Ah! Sorry, I've somehow missed that lambda.

Since we don't change `in` (and thus `end(in)` always returns the same pointer), I'd probably just use:

```cpp
auto i = in.data();
auto e = i + in.size();
...
```

Reply:
Fixed